fix(fs/ensure_symlink): check symlink is pointing the given target

ryota2357 opened the pull request:
fix: #4363

> So the API doc seems correct, and I think we should check the symlink is actually pointing the given target file if `Deno.errors.AlreadyExists` is thrown

I implemented it.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/denoland/deno_std?pullRequest=4371) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/denoland/deno_std?pullRequest=4371) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/denoland/deno_std?pullRequest=4371) it.</sub>

kt3k commented:
This solution replaces the link with a new link when the link target is not correct. I think this is one good solution, but I also wonder if it makes sense to throw some error when the existing link target is not correct. What do you think?

ryota2357 commented:
> throw some error when the existing link target is not correct.
- I feels from "ensureSymlink(A, B)" word that "B always links to A"
- Anothoer ensure* function such that "ensureDir()" create directory at a given path.

From these, I think that there is no need to throw an error if the existing link target is not correct, only make the correct symlink.

However, I do not know enough about English or API design to know which is better.

kt3k commented:
> Anothoer ensure* function such that "ensureDir()" create directory at a given path.

Yes, they create new dirs or files, but they don't overwrite or replace existing things to my understanding. However this change replaces the existing symlink with a new one, and that symlink is silently lost in their system.

I personally think throwing some error with the message like `A symlink targeting to an undesired path already exists: ${linkName} -> ${linkRealPath}` would make more sense.

ryota2357 commented:
I apologize for late response.

> they don't overwrite or replace existing things

I see, I think this is also clear and looks good to me.

I updated the implementation and added doc.
Could you please re-review it?

kt3k reviewed (approved):
LGTM. Thanks for the fix!
